### ISSUE TITLE
question about the finetune

### ISSUE THREAD

**Issue by LiMa-cas** [Vahe1994/AQLM#127]
1. is the finetune need each layer？ could I used for some layers finetune once？
2. is codebook quantized method is slower than AWQ?
3. when I inference，it is successful when max_new_tokens=512， but failed when max_new_tokens=2048

**Comment by Vahe1994** [comment]
Hi @LiMa-cas,

1. I'm not sure I fully understand you. In the code, we have both block fine-tuning and global fine-tuning. You can do just block fine-tuning, which fine-tunes each layer once, but the results will likely be worse compared to doing global fine-tuning on top of it.
  
2. If you’re referring to the quantization phase, yes, it is slower than AWQ or GPTQ (please see [this link](https://github.com/Vahe1994/AQLM?tab=readme-ov-file#quantization-time) for more details on quantization time). However, you only need to perform quantization once, and afterward, you can reuse your quantized model during inference as often as you like (please see [inference details](https://github.com/Vahe1994/AQLM?tab=readme-ov-file#inference-kernels) for more information on inference time).

3. Can you please post the error you get when using `max_new_tokens=2048`? Also, how much VRAM do you have? Are you running into an out-of-memory error?

**Comment by LiMa-cas** [comment]
Thanks very much. I might not have expressed myself clearly last time.
1. why block fine-tuning is worse than global  fine-tuning?
2. the inference time instead of quantization is much slower than AWQ in GPU (A6000,50G)，maybe because the codebook？
3. it’s on A6000,50G, when I using max_new_tokens=2048，After loading the model, the memory usage keeps increasing up to 40G and decreasing down to 8G continuously and could not run successfullly ； but it is successful on max_new_tokens=512;

**Comment by Godofnothing** [comment]
@LiMa-cas 
1) As discussed in the paper, global fine-tuning is a more expressive (and more expensive) finetuning. One optimizes the target loss directly instead of per-block L2 loss, which is correlated with performance but related to it in a complicated way.
2) AWQ and any other scalar quantization is likely to be faster than AQLM since weight dequantization is simpler. AQLM doesn't offer superior inference speed to scalar quantization, the main value proposition is better performance at extreme compression (2 and below bits), where scalar quantization methods (GPTQ, AWQ) yield gibberish text.

**Comment by LiMa-cas** [comment]
Thanks a lot！

**Comment by LiMa-cas** [comment]
![{6D34503E-D6A0-4BB6-B0A3-674A14EFE1A2}](https://github.com/user-attachments/assets/96d5d817-12a5-45c9-a1ce-bd38d8ea1496)
Hi， when I use pv-tuning for AWQ model，firstly I need to 
![{6D34503E-D6A0-4BB6-B0A3-674A14EFE1A2}](https://github.com/user-attachments/assets/3f2a4b26-59f0-42c4-8e60-7ccb42998cba)
but I encountered a bug：
![{121A4318-B370-402F-A59E-FD7BC4E5845E}](https://github.com/user-attachments/assets/15c7a5d3-8e8c-4d9d-98bc-319a387d2e48)
could u help me？I used --load_dtype float16 instead

**Comment by Godofnothing** [comment]
Hi, @LiMa-cas.
Sorry for the delayed reply.
The issue seems to be with dataset loading. We have not encountered such an error. Probably it is due to the internet connection.
Does it reproduce on another launch of the script?
